### PR TITLE
Bump version of `lsp` library

### PR DESCRIPTION
**Pull request by emilienlemaire**
As of now the tests do not pass, the diffs are as follow, I have yet to understand why:
```diff
File "test/lsp/lsp_formatting.ml", line 1, characters 0-0:
diff --git a/_build/default/test/lsp/lsp_formatting.ml b/_build/default/test/lsp/lsp_formatting.ml.corrected
index bad67110..b836af07 100644
--- a/_build/default/test/lsp/lsp_formatting.ml
+++ b/_build/default/test/lsp/lsp_formatting.ml.corrected
@@ -50,7 +50,7 @@ let%expect_test "simple-formatting-request" =
   end_with_postproc [%expect.output];
   [%expect {|
     {"params":{"diagnostics":[],"uri":"file://__rootdir__/prog.cob"},"method":"textDocument/publishDiagnostics","
jsonrpc":"2.0"}
-    PROGRAM-ID. HELLO.
+           PROGRAM-ID. HELLO.
     PROCEDURE DIVISION.
     para-1.
         DISPLAY "HELLO"
@@ -73,7 +73,7 @@ let%expect_test "unindent-formatting-request" =
   end_with_postproc [%expect.output];
   [%expect {|
     {"params":{"diagnostics":[],"uri":"file://__rootdir__/prog.cob"},"method":"textDocument/publishDiagnostics","
jsonrpc":"2.0"}
-    PROGRAM-ID. HELLO.
+           PROGRAM-ID. HELLO.
     PROCEDURE DIVISION.
     para-1.
         DISPLAY "HELLO"
@@ -102,7 +102,7 @@ let%expect_test "formatting-request-nested-if" =
   end_with_postproc [%expect.output];
   [%expect {|
     {"params":{"diagnostics":[{"message":"Invalid syntax","range":{"end":{"character":16,"line":1},"start":{"char
acter":7,"line":1}},"severity":1}],"uri":"file://__rootdir__/prog.cob"},"method":"textDocument/publishDiagnostics"
,"jsonrpc":"2.0"}
-    PROCEDURE DIVISION.
+           PROCEDURE DIVISION.
     para-1.
         IF X>9
         THEN
@@ -138,7 +138,7 @@ let%expect_test "formatting-request-data" =
   end_with_postproc [%expect.output];
   [%expect{|
     {"params":{"diagnostics":[{"message":"Invalid syntax","range":{"end":{"character":23,"line":1},"start":{"char
acter":8,"line":1}},"severity":1}],"uri":"file://__rootdir__/prog.cob"},"method":"textDocument/publishDiagnostics"
,"jsonrpc":"2.0"}
-    WORKING-STORAGE SECTION.
+            WORKING-STORAGE SECTION.
         01 x.
         05 y.
         10 z pic 999.
@@ -204,7 +204,7 @@ let%expect_test "formatting-request-nested-program" =
   end_with_postproc [%expect.output];
   [%expect{|
     {"params":{"diagnostics":[],"uri":"file://__rootdir__/prog.cob"},"method":"textDocument/publishDiagnostics","
jsonrpc":"2.0"}
-    IDENTIFICATION DIVISION.
+            IDENTIFICATION DIVISION.
     PROGRAM-ID. X.
     PROCEDURE DIVISION.
         DISPLAY "I'm in X"
@@ -256,7 +256,7 @@ let%expect_test "formatting-request-alignment-argument" =
   end_with_postproc [%expect.output];
   [%expect {|
     {"params":{"diagnostics":[{"message":"Invalid syntax","range":{"end":{"character":16,"line":1},"start":{"char
acter":7,"line":1}},"severity":1}],"uri":"file://__rootdir__/prog.cob"},"method":"textDocument/publishDiagnostics"
,"jsonrpc":"2.0"}
-    PROCEDURE DIVISION.
+           PROCEDURE DIVISION.
         MOVE VAR-1 TO VAR-2 VAR-3
                       VAR-4
                       VAR-5. |}]
@@ -283,7 +283,7 @@ let%expect_test "formatting-request-else-if" =
   end_with_postproc [%expect.output];
   [%expect {|
     {"params":{"diagnostics":[{"message":"Invalid syntax","range":{"end":{"character":16,"line":1},"start":{"char
acter":7,"line":1}},"severity":1}],"uri":"file://__rootdir__/prog.cob"},"method":"textDocument/publishDiagnostics"
,"jsonrpc":"2.0"}
-    PROCEDURE DIVISION.
+           PROCEDURE DIVISION.
         if x>1
            move 1 to x
         else if x>2
@@ -395,7 +395,7 @@ let%expect_test "formatting-request-whole-program" =
   end_with_postproc [%expect.output];
   [%expect {|
     {"params":{"diagnostics":[],"uri":"file://__rootdir__/prog.cob"},"method":"textDocument/publishDiagnostics","
jsonrpc":"2.0"}
-    IDENTIFICATION DIVISION.
+            IDENTIFICATION DIVISION.
     PROGRAM-ID. MACESDS.
     ENVIRONMENT DIVISION.
     INPUT-OUTPUT SECTION.
@@ -506,7 +506,7 @@ let%expect_test "formatting-request-on-exception" =
   end_with_postproc [%expect.output];
   [%expect{|
     {"params":{"diagnostics":[{"message":"Invalid syntax","range":{"end":{"character":16,"line":1},"start":{"char
acter":7,"line":1}},"severity":1}],"uri":"file://__rootdir__/prog.cob"},"method":"textDocument/publishDiagnostics"
,"jsonrpc":"2.0"}
-    PROCEDURE DIVISION.
+           PROCEDURE DIVISION.
         CALL STH
            NOT ON EXCEPTION
               RAISE EXCEPTION exception-name-1
@@ -535,7 +535,7 @@ let%expect_test "formatting-request-perform" =
   end_with_postproc [%expect.output];
   [%expect {|
     {"params":{"diagnostics":[{"message":"Invalid syntax","range":{"end":{"character":16,"line":1},"start":{"char
acter":7,"line":1}},"severity":1}],"uri":"file://__rootdir__/prog.cob"},"method":"textDocument/publishDiagnostics"
,"jsonrpc":"2.0"}
-    PROCEDURE DIVISION.
+           PROCEDURE DIVISION.
     para-1.
         PERFORM 3 TIMES
            PERFORM PARA-2
```

These errors appear after bumping `lsp` and `jsonrpc` to version `>=1.17` (although I did not test for version `1.16`)

### CODE REVIEW

File: src/lsp/cobol_parser/post/gen_post_actions.ml [hunk 1]
```diff
@@ -141,7 +141,7 @@ let pp_production_posts ppf =
       | func -> Fmt.pf ppf "%s (@[<2>(%a)@ value@])" tag Fmt.text func
     in
     match post_action (last_item_attrs p),
-          post_action (Production.attributes p) with
+          post_action (Nonterminal.attributes (Production.lhs p)) with
```
Comment (nberth) [inline src/lsp/cobol_parser/post/gen_post_actions.ml:144]: Could you check the generated file is the same with this change?

**Review by nberth — COMMENTED** [review]
To investigate the error, I suggest to split the bump of `lsp` version and the update of `merlin` into separate commits (or better: separate PRs). Also the latter should relax the constraint on `menhir` (we have a `menhir = "<some specific version>"` somewhere in a `package.toml`, probably in `cobol_parser`).

File: src/lsp/cobol_parser/post/gen_post_actions.ml (hunk 1 above)
Comment (emilienlemaire) [reply src/lsp/cobol_parser/post/gen_post_actions.ml:144]: The generated files are exactly the same

File: src/lsp/cobol_parser/package.toml [hunk 2]
```diff
@@ -67,7 +67,6 @@ version = ">=1.2"
 # package tools dependencies
 [tools]
 ppx_deriving = ">=5.2.1"
-menhir = "=20230415"
```
Comment (nberth) [inline src/lsp/cobol_parser/package.toml:70]: We probably need a lower-bound on the version, now.

**Comment by emilienlemaire** [comment]
This PR is on hold until the next release of [ocaml-lsp](https://github.com/ocaml/ocaml-lsp), which is needed for a bug fix.

**Comment by nberth** [comment]
> This PR is on hold until the next release of [ocaml-lsp](https://github.com/ocaml/ocaml-lsp) `>1.17.0`, which is needed for a bug fix.

[Looks like](https://github.com/ocaml/opam-repository/pull/26202) it's imminent ;-)

**Comment by nberth** [comment]
Closed in favor or more recent #321

**Pull request closed without merging**